oc_cdtapi/ForemanAPI.py

```diff
@@ -301,861 +288,823 @@ def create_host_v2 (self, task, custom_json):
-            return self.get_report_v1 (id)
-        elif self.apiversion ==  2:
-            logging.debug ('Passing to get_report_v2')
-            return self.get_report_v2 (id)
-
+            logging.debug('Passing to get_report_v1')
+            return self.get_report_v1(id)
+        elif self.apiversion == 2:
+            logging.debug('Passing to get_report_v2')
+            return self.get_report_v2(id)
 
-    def get_report_v1 (self, id):
+    def get_report_v1(self, id):
         """
         Returns a foreman report with specified id
         :param id: int, id of the required report
         :return: response in json format
         """
-        response = self.get("config_reports/" + str(id), headers = self.headers)
+        response = self.get(posixpath.join("config_reports", str(id)), headers=self.headers)
         return response.json()
 
-
-    def get_report_v2 (self, id):
+    def get_report_v2(self, id):
         """
         Returns a foreman report with specified id
         :param id: int, id of the required report
         :return: response in json format
         """
-        logging.debug ('Reached get_report_v2')
-        logging.debug ('Passing to get_report_v1')
-        return self.get_report_v1 (id)
+        logging.debug('Reached get_report_v2')
+        logging.debug('Passing to get_report_v1')
+        return self.get_report_v1(id)
 
-    
-    def get_hostgroup_id (self, hostgroup_name):
+    def get_hostgroup_id(self, hostgroup_name):
         """
         wrapper for api v1/v2
         """
-        logging.debug ('Reached get_hostgroup_id')
+        logging.debug('Reached get_hostgroup_id')
         if self.apiversion == 1:
-            logging.debug ('Passing to get_hostgroup_id_v1')
-            return self.get_hostgroup_id_v1 (hostgroup_name)
+            logging.debug('Passing to get_hostgroup_id_v1')
+            return self.get_hostgroup_id_v1(hostgroup_name)
         elif self.apiversion == 2:
-            logging.debug ('Passing to get_hostgroup_id_v2')
-            return self.get_hostgroup_id_v2 (hostgroup_name)
-
+            logging.debug('Passing to get_hostgroup_id_v2')
+            return self.get_hostgroup_id_v2(hostgroup_name)
 
-    def get_hostgroup_id_v1 (self, hostgroup_name):
+    def get_hostgroup_id_v1(self, hostgroup_name):
         """
         Returns id of the required hostgroup
         :param hostgroup_name: str
         :return: int
         """
-        logging.debug ('Reached get_hostgroup_id_v1')
-        hostgroups = self.get("hostgroups", headers = self.headers).json()["results"]
-        for hostgroup in hostgroups:
-            if hostgroup["name"] == hostgroup_name:
-                return hostgroup.get ('id')
+        logging.debug('Reached get_hostgroup_id_v1')
+        hostgroups = self.get("hostgroups", headers=self.headers).json()["results"]
+        hostgroups = list(filter(lambda x: x.get("name")==hostgroup_name, hostgroups))
```

Comment:
This writing does not add readability and complicates code maintenance

Reply:
Disagree that three-lined loop with different indent is more readable than one simple `lambda filter`.
But reverted to (almost) previous version.